File: FleetMaintenance.Domain/Entities/MaintenanceRecord.cs
using FleetMaintenance.Domain.Enums;

namespace FleetMaintenance.Domain.Entities;

public class MaintenanceRecord
{
    public int Id { get; set; }

    public int VehicleId { get; set; }

    public int MaintenanceTypeId { get; set; }

    public DateTime ScheduledDate { get; set; }

    public DateTime? CompletedDate { get; set; }

    public int? MileageAtService { get; set; }

    public int? DueMileage { get; set; }

    public decimal? Cost { get; set; }

    public string? Notes { get; set; }

    public MaintenanceStatus Status { get; set; }
        = MaintenanceStatus.Scheduled;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public Vehicle Vehicle { get; set; } = null!;

    public MaintenanceType MaintenanceType { get; set; } = null!;
}

File: fleet-maintenance-client/src/types/maintenanceRecord.ts
export type MaintenanceStatus =
  | "Scheduled"
  | "Completed"
  | "Cancelled";

export interface MaintenanceRecord {
  id: number;
  vehicleId: number;
  vehiclePlateNumber: string;
  maintenanceTypeId: number;
  maintenanceTypeName: string;
  scheduledDate: string;
  completedDate: string | null;
  mileageAtService: number | null;
  dueMileage: number | null;
  cost: number | null;
  notes: string | null;
  status: MaintenanceStatus;
  isOverdue: boolean;
  createdAt: string;
}

export interface MaintenanceRecordFilters {
  search: string;
  vehicleId: number | null;
  maintenanceTypeId: number | null;
  status: MaintenanceStatus | "";
  fromDate: string;
  toDate: string;
  pageNumber: number;
  pageSize: number;
}

export interface CreateMaintenanceRecordRequest {
  vehicleId: number;
  maintenanceTypeId: number;
  scheduledDate: string;
  dueMileage?: number | null;
  notes?: string | null;
}

export interface UpdateMaintenanceRecordRequest {
  maintenanceTypeId?: number;
  scheduledDate?: string;
  dueMileage?: number;
  notes?: string;
}

export interface CompleteMaintenanceRecordRequest {
  completedDate: string;
  mileageAtService: number;
  cost: number;
  notes?: string | null;
}